src/mds/Server.cc

```diff
@@ -5955,6 +5955,49 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::make_quota_max_byte_parseable(string val)
+{
+  string before_decimal;
+  string after_decimal;
+  int64_t dec_len;
+  int64_t q;
+  string cast_err;
+  
+  size_t dec_index = val.find_first_of(".");
+  size_t unit_index = val.find_first_not_of("0123456789-+.");
+  unit = str.substr(unit_index, val.length() - unit_index);
+  if (dec_index != std::string::npos) {
+    if ((unit_index == std::string::npos) || (unit == 'B'))  {
+    dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
+    << "Invalid Byte value" << dendl;
+    return -CEPHFS_EINVAL;
+    }
+  }
+  dec_len = unit_index - dec_index - 1;
+  if ((dec_len == 0) || (dec_index == val.length() - 1)) {
+    dout(10) << __func__ << ":  failed to parse quota.max_bytes: "
+    << "Invalid Float value" << dendl;
+    return -CEPHFS_EINVAL;
+  }  
```

Comment:
The order is kept this way intentionally, because the check for `unit_index` is further required by second order of checks. It would be nice if we stick with this